### PR TITLE
Add docs generation script.

### PR DESCRIPTION
**Pull request by Lukasa**
Motivation:

Documentation is a real help when using a project, we should be
generating it.

Modifications:

- Ported docs script from SwiftNIO.

Result:

We can provide API documentation.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -26,7 +26,7 @@ RUN mkdir -p $HOME/.tools
 RUN echo 'export PATH="$HOME/.tools:$PATH"' >> $HOME/.profile
 
 # script to allow mapping framepointers on linux (until part of the toolchain)
-RUN wget -q https://raw.githubusercontent.com/apple/swift/master/utils/symbolicate-linux-fatal -O $HOME/.tools/symbolicate-linux-fatal
+RUN wget -q https://raw.githubusercontent.com/apple/swift/main/utils/symbolicate-linux-fatal -O $HOME/.tools/symbolicate-linux-fatal
 RUN chmod 755 $HOME/.tools/symbolicate-linux-fatal
```
Comment (tomerd) [inline docker/Dockerfile:30]: we can probably just get rid of this all-together, we do not need it

File: scripts/generate_docs.sh [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+set -ex
+
+my_path="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+root_path="$my_path/.."
+version=$(git describe --abbrev=0 --tags || echo "0.0.0")
+modules=( NIOSSH )
```
Comment (tomerd) [inline scripts/generate_docs.sh:21]: I see there a re a f ew other modules in the repo, is including only NIOSSH in the docs intentional?

**Review by tomerd — APPROVED** [review]
looks good, but a couple of questions

File: scripts/generate_docs.sh (hunk 2 above)
Comment (Lukasa) [reply scripts/generate_docs.sh:21]: Yes, the other modules are binaries.

File: docker/Dockerfile (hunk 1 above)
Comment (Lukasa) [reply docker/Dockerfile]: Done.

**Pull request merged**